src/utils/dataAttributes.ts

```diff
@@ -1,28 +1,80 @@
 export const DataAttribute = {
   ElementCodename: 'data-kontent-element-codename',
+  ComponentId: 'data-kontent-component-id',
   ItemId: 'data-kontent-item-id',
   LanguageCodename: 'data-kontent-language-codename',
   ProjectId: 'data-kontent-project-id',
 };
 
+const DataAttributeHierarchy = [
+  DataAttribute.ProjectId,
+  DataAttribute.LanguageCodename,
+  DataAttribute.ItemId,
+  DataAttribute.ComponentId,
+  DataAttribute.ElementCodename,
+];
+
+const DataAttributeParsingOrder = DataAttributeHierarchy.slice().reverse();
+const OptionalDataAttributes = [DataAttribute.ComponentId];
+
+function getHigherDataAttributes(attributeName: string): ReadonlyArray<string> {
+  const attributeIndex = DataAttributeHierarchy.indexOf(attributeName);
+  return DataAttributeHierarchy.slice(0, attributeIndex);
+}
+
 export function dataToDatasetAttributeName(dataAttributeName: string): string {
   return dataAttributeName.replace(/^data-/, '').replace(/-./g, (x: string) => x.toUpperCase()[1]);
 }
 
-export function getDataAttributesFromEventPath(event: Event): Map<string, string> {
+export function getDataAttributesFromEventPath(event: Event): ReadonlyMap<string, string> {
   const path = event.composedPath();
   const parsed: Map<string, string> = new Map();
 
   for (const node of path) {
     if (!(node instanceof HTMLElement)) continue;
 
-    for (const [, attrName] of Object.entries(DataAttribute)) {
-      const datasetAttrName = dataToDatasetAttributeName(attrName);
-      const datasetAttr = node.dataset[datasetAttrName];
+    for (const attributeName of DataAttributeParsingOrder) {
+      const datasetAttributeName = dataToDatasetAttributeName(attributeName);
+      const datasetAttributeValue = node.dataset[datasetAttributeName];
+      const isAttributeOptional = OptionalDataAttributes.includes(attributeName);
 
-      if (datasetAttr && !parsed.has(attrName)) {
-        parsed.set(attrName, datasetAttr);
+      if (!datasetAttributeValue || parsed.has(attributeName)) continue;
+
+      // Higher attributes are attributes that are placed higher in the hierarchy, for example,
+      // for component id attribute those will be item id, language codename and project id.
+      const higherDataAttributes = getHigherDataAttributes(attributeName);
+      const areSomeHigherAttributesSet = higherDataAttributes.some((attr) => parsed.has(attr));
```

Comment:
Unfortunately, this doesn't work, because has method loses its `this` context in that case.
![image](https://user-images.githubusercontent.com/57262755/91699664-e5103600-eb74-11ea-90a7-46b22b447d53.png)